core/controllers/acl_decorators.py

```diff
@@ -2753,6 +2753,84 @@ def test_can_accept_suggestion(
     return generate_decorator_for_handler
 
 
+def can_update_suggestions(handler):
+    """Decorator to check whether user can update the suggestions that
+    they are allowed to review.
+
+    Args:
+        handler: function. The function to be decorated.
+
+    Returns:
+        function. The newly decorated function that has common checks and
+        permissions specified by passed in decorator. Users with rights
+        to accept translations andthe user who created a particular
+        translation are allowed to perform this action.
+
+    Raises:
+        NotLoggedInException. The user is not logged in.
+        UnauthorizedUserException. The user does not have credentials to
+            edit this suggestion.
+        InvalidInputException. The submitted suggestion id is not valid.
+        PageNotFoundException. A suggestion is not found with the given
+            suggestion id.
+    """
+    def test_can_update_suggestion(
+            self, suggestion_id, **kwargs):
+        """Returns a handler to test whether a
+        suggestion can be updated based on the user roles.
+
+        Args:
+            suggestion_id: str. The suggestion id.
+            **kwargs: *. Keyword arguments.
+
+        Returns:
+            function. The handler for updating a
+            suggestion.
+
+        Raises:
+            NotLoggedInException. The user is not logged in.
+            UnauthorizedUserException. The user does not have credentials to
+                edit this suggestion.
+            InvalidInputException. The submitted suggestion id is not valid.
+            PageNotFoundException. A suggestion is not found with the given
+                suggestion id.
+        """
+        if not self.user_id:
+            raise base.UserFacingExceptions.NotLoggedInException
+        user_actions = user_services.get_user_actions_info(
+            self.user_id
+        ).actions
+        if role_services.ACTION_ACCEPT_ANY_SUGGESTION in user_actions:
+            return handler(self, suggestion_id, **kwargs)
+
+        if len(suggestion_id.split('.')) != 3:
+            raise self.InvalidInputException(
+                'Invalid format for suggestion_id.'
+                ' It must contain 3 parts separated by \'.\'')
```

Comment:
Does ``get_suggestion_by_id`` check the ID format/do we need to check it again here?

Reply:
It's not checked in the suggestion service layer. Hence I think we need to add the check here.

Reply:
@nithusha21 do you have any thoughts on this? Is it expected at the controller level for all interactions with ``get_suggestion_by_id`` first check that the ID is of the valid suggestion ID format?